src/plugins/intel_gna/src/runtime/cnn.cpp

```diff
@@ -317,8 +317,11 @@ bool is2D(T&& vec) {
 }
 }  // namespace
 
-void CNNMaxPool(intel_dnn_component_t* component, intel_dnn_number_type_t number_type, const bool sumPoolingOverRide) {
-    if (is2D(component->op.maxpool.poolingStrideXY) || is2D(component->op.maxpool.poolingWindowXY)) {
+void CNNMaxPool(intel_dnn_component_t* component,
+                intel_dnn_number_type_t number_type,
+                const bool enforce_2d,
+                const bool sumPoolingOverRide) {
+    if (enforce_2d || is2D(component->op.maxpool.poolingStrideXY) || is2D(component->op.maxpool.poolingWindowXY)) {
```

Comment:
Cannot we just detect the target instead of pushing some artificial op type through many levels of interface only for the SW_FP32 mode?

Reply:
good Idea , I hope I got your point and new approach meets expectations.